File: src/main/java/nl/mpcjanssen/simpletask/AlarmReceiver.java
package nl.mpcjanssen.simpletask;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import nl.mpcjanssen.simpletask.dao.gen.LogItemDao;


import java.util.Date;

public class AlarmReceiver extends BroadcastReceiver {
    private static final String TAG = "AlarmReceiver";

    @Override
    public void onReceive(Context context, Intent intent) {
        Logger.INSTANCE.info(TAG, "Executing Alarm callback");
        TodoApplication m_app = (TodoApplication) context.getApplicationContext();

        // Clean up logging
        Date now = new Date();
        Date removeBefore = new Date(now.getTime()-24*60*60*1000);
        long oldLogCount = m_app.logDao.count();
        m_app.logDao.queryBuilder()
                .where(LogItemDao.Properties.Timestamp.lt(removeBefore))
                .buildDelete()
                .executeDeleteWithoutDetachingEntities();
        long logCount = m_app.logDao.count();
        Logger.INSTANCE.info(TAG, "Cleared " + (oldLogCount-logCount) + " old log items");

        // Update UI (widgets and main screen)
        m_app.getLocalBroadCastManager().sendBroadcast(new Intent(Constants.BROADCAST_UPDATE_UI));
    }
}

File: app/src/main/java/nl/mpcjanssen/simpletask/AlarmReceiver.kt
package nl.mpcjanssen.simpletask

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import android.util.Log
import nl.mpcjanssen.simpletask.task.TodoList
import nl.mpcjanssen.simpletask.util.broadcastFileSync

class AlarmReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val lbm = LocalBroadcastManager.getInstance(context)
        Log.i(TAG, "Executing Alarm callback")
        if (Constants.ALARM_RELOAD == intent.getStringExtra(Constants.ALARM_REASON_EXTRA)) {
            broadcastFileSync( lbm )
        } else {
            TodoApplication.todoList.reload("Alarm")
        }

    }

    companion object {
        private val TAG = AlarmReceiver::class.java.simpleName
    }
}
